atat/src/ingress_manager.rs

```diff
@@ -1,9 +1,42 @@
-use heapless::{consts, ArrayLength, String};
+use heapless::{consts, ArrayLength, Vec};
 
 use crate::error::Error;
 use crate::queues::{ComConsumer, ResProducer, UrcProducer};
 use crate::{Command, Config};
 
+use core::iter::FromIterator;
+
+trait SliceExt {
+    fn trim(&self, whitespaces: &[u8]) -> &Self;
+    fn trim_start(&self, whitespaces: &[u8]) -> &Self;
+}
+
+impl SliceExt for [u8] {
+    fn trim(&self, whitespaces: &[u8]) -> &[u8] {
+        let is_not_whitespace = |c| !whitespaces.contains(c);
+
+        if let Some(first) = self.iter().position(is_not_whitespace) {
+            if let Some(last) = self.iter().rposition(is_not_whitespace) {
+                &self[first..=last]
+            } else {
+                unreachable!();
+            }
+        } else {
+            &[]
+        }
+    }
+
+    fn trim_start(&self, whitespaces: &[u8]) -> &[u8] {
+        let is_not_whitespace = |c| !whitespaces.contains(c);
+
+        if let Some(first) = self.iter().position(is_not_whitespace) {
+            &self[first..]
+        } else {
+            &[]
+        }
+    }
+}
```

Comment:
This might be an easy candidate for some unit tests :slightly_smiling_face:

Reply:
True! Will add a few to check the logic